Dev steven

SteveLaird opened the pull request:
Testing pull requests.

dyfries committed:
Breaking the object reference demo into its own scene. Cleaning up scene for clarity.

dyfries committed:
Import Schmupp demo project

dyfries committed:
Merge branch 'develop' into Dev_Dylan

# Conflicts:
#	Assets/2DSchmupp.meta
#	Assets/2DSchmupp/Art.meta
#	Assets/2DSchmupp/Art/Animations.meta
#	Assets/2DSchmupp/Art/Animations/Effects.meta
#	Assets/2DSchmupp/Assets.meta
#	Assets/2DSchmupp/Assets/Effects.meta
#	Assets/2DSchmupp/Assets/Effects/enemy-death.meta
#	Assets/2DSchmupp/Assets/Player And NPCs.meta
#	Assets/2DSchmupp/Assets/Player And NPCs/_NPCs.meta
#	Assets/2DSchmupp/Code.meta
#	Assets/2DSchmupp/Code/NPC_Spawner.cs
#	Assets/2DSchmupp/Code/SimpleCollider2D.cs
#	Assets/2DSchmupp/Prefabs.meta
#	Assets/2DSchmupp/Prefabs/NPCs.meta
#	Assets/2DSchmupp/Prefabs/Player.meta
#	Assets/2DSchmupp/Prefabs/Player/Bullets.meta
#	Assets/2DSchmupp/Prefabs/Spawners.meta

Tsurei committed:
Merge branch 'develop' into Dev_Russ

Tsurei committed:
Revert "Create Shadow.gif"

This reverts commit ad532aaba3272735984d88470bfb40774baf3614.

Tsurei committed:
Merge branch 'develop' into Dev_Russ

Tsurei committed:
Merge branch 'develop' into Dev_Russ

Tsurei committed:
Revert "Revert "Create Shadow.gif""

This reverts commit d69d28fd073d0a985ae9363756a066b7caa522dc.

Tsurei committed:
Create Shadow.gif.meta

dyfries committed:
Minor comments about doing the spawner breakdown.

CraigFranceschetti committed:
added working document for the team to view. Would like to push this up so that the team has access.

Tsurei committed:
Added Npc script with stubs and some comments

CraigFranceschetti committed:
Added the initial code stubs for the spawn wave settings.

OlivierRRC committed:
Merge pull request #13 from dyfries/Dev_Craig

Initial Code for SpawnWaveSettings

dyfries committed:
Some enum examples and other misc (settings, adding stubs to NPC, settings default build scene, build settings. ).

dyfries committed:
Merge branch 'develop' of https://github.com/dyfries/SharedGameP1_23 into develop

OlivierRRC committed:
Commented WaveSettings and  added wave speed variable

SteveLaird committed:
Day 1. NPC_Spawner initial changes to Start and Update methods and variables added.

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Russ' into mergingBranch develop

# Conflicts:
#	Assets/2DSchmupp/Code/BaseNPC.cs
#	Assets/2DSchmupp/Code/NPC_Spawner.cs

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Olivier' into mergingBranch

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Steven' into mergingBranch

dyfries committed:
Merge branch 'develop' into Dev_Russ

dyfries reviewed (changes requested):
Don't merge into main

Tsurei committed:
Merge branch 'develop' into Dev_Russ

Tsurei committed:
Changed NPC Movement script

Tsurei committed:
Merge branch 'develop' into Dev_Russ

Tsurei committed:
Merge branch 'Dev_Russ' of https://github.com/dyfries/SharedGameP1_23 into Dev_Russ

Tsurei committed:
Update NpcMovement and move into 2DSchmupp folder

Tsurei committed:
Removed file as it was moved to another folder

dyfries committed:
Basic Raycasting demo

dyfries committed:
More elaborate example using a Contact Filter.

dyfries committed:
Made Example object a prefab

dyfries committed:
Created a scriptable object wave settings and changed the prefab

dyfries committed:
Made changes in the map: Changed boundary color

dyfries committed:
Created new prefab for demo

CraigFranceschetti committed:
Minor refactoring/commenting of code.

Tsurei committed:
Update NpcMovement.cs

Tsurei committed:
Added Comments declaring some methods are stubs that will be revamped in future

LorenzoOrtizRRC committed:
- Refactored timings.
- NPC_Spawner now iterates through WaveSettings.
- Created new test scene for Lorenzo.

LorenzoOrtizRRC committed:
- Renamed Lorenzo's test scene.

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Craig' into mergingBranch

dyfries committed:
Commit new version of the NPC, renamed from NpcMovement to NPCMovement, Created some sample assets and setup a folder for this spawner project.

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Lorenzo' into mergingBranch

dyfries committed:
Spawner Dev branch merged in and some minor renaming / organizing.

CraigFranceschetti committed:
Single NPCMovement instead of array. Created local offset, and wave speed. Accessed flight patterns from NPC and serialized it.

CraigFranceschetti committed:
Merge pull request #18 from dyfries/UpdatedWaveSettings

Simplified and expanded WaveSettings

Tsurei committed:
Changed speed variable to a float, adjusted linear movement

Tsurei committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into mergingBranch

SteveLaird committed:
Unfinished Changes to spawner script

SteveLaird committed:
Merge branch 'mergingBranch' into Dev_Steven

SteveLaird committed:
small change to spawner script

CraigFranceschetti committed:
Added Snek NPC to be spawned.

CraigFranceschetti committed:
Merge pull request #19 from dyfries/Snek

Added Snek NPC to be spawned.

Tsurei committed:
Updated Snek sprite

SteveLaird committed:
Spawner Team Changes

SteveLaird committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into mergingBranch

dyfries committed:
Debugging Notes

dyfries committed:
Ability Class Demos

dyfries committed:
Polymorphism and subclasses demo

dyfries committed:
Stats and Cards Prefab settings demo

dyfries committed:
Updated (mostly commented) the ability component - This is v2
Updated Player Prefab to have ability component. Updated main scene to have player prefab with Lazer.
Updated Demos scenes
Removed Lazer from base player 00
Created Player 01 with the Lazer

V2 Ability Lazer Asset Export.

Created a shared Lesson Bundles folder since this asset is in many different classes right now.

dyfries committed:
Moved Devs folders into their own dedicated place and out of the way.

dyfries committed:
Import Ability V3 Asset. This includes art and animation examples.

dyfries committed:
Updates to Ability system for Subclass ability exercise.
Ability_Simple is now abstract, removed all art and physics references.

LorenzoOrtizRRC committed:
- Added Ability_StarWeaving script.

LorenzoOrtizRRC committed:
- Added basic fields.

LorenzoOrtizRRC committed:
- Added base implementation of Star Weaving ability.

LorenzoOrtizRRC committed:
- Finished base implementation of ability coroutine.

LorenzoOrtizRRC committed:
- Created new scene to test ability.
- Created new Projectile_StarWeavingOrb script.

dyfries committed:
In progress refactor of the Simple Ability (Commented code still there)
Created subclasses for boost and huge Lazer abilities
Added more art to the player for use with the speed boost.
Added collision tags to the project (but not individual units yet)

LorenzoOrtizRRC committed:
- Finished base implementation of Projectile_StarWeavingOrb.

dyfries committed:
Old is just there for reference (for now), removed all the commented out code from Ability Simple

OlivierRRC committed:
Added missile barrage script

Created my own scene for testing
created Missile prefab to be spawned
Created missile barrage script which shoots out a barrage of missiles

dyfries committed:
Added physics to Huge Lazer ability (got lost in refactoring), saved scene and prefabs.

Tsurei committed:
Added blink(teleport) ability to Dev_Russ

LorenzoOrtizRRC committed:
- Created prefabs for ability and projectile.

LorenzoOrtizRRC committed:
- Refactored ability.

LorenzoOrtizRRC committed:
- Ability now sets projectiles' directions.
- Fixed reference bugs from previous refactoring.

LorenzoOrtizRRC committed:
- Fixed the projectiles' directions to be transform.up instead of transform.forward.

Tsurei committed:
Removed unnecessary code, added comments

LorenzoOrtizRRC committed:
- Set up test scene for ability and projectile.

alexbacon01 committed:
started a slow down ability using the base ability class -

Tsurei committed:
Removed unnecessary code of Transform

agamis28 committed:
Added invinsibility ability sub class from base class 'ability_simple'

LorenzoOrtizRRC committed:
- Finished initial balancing of ability and projectile settings.

CraigFranceschetti committed:
Added Bomb ability. Currently just spawns in the middle. Need to change physics layers so that it doesn't affect the player. Did not want to commit my physics layer settings to avoid a merge conflict.

LorenzoOrtizRRC committed:
- edited test scene variables.

agamis28 committed:
Merge remote-tracking branch 'origin/develop' into Dev_Angel

LorenzoOrtizRRC committed:
Merge branch 'develop' into Dev_Lorenzo

Tsurei committed:
Utilizing rigid built in method

Tsurei committed:
Merge branch 'develop' into Dev_Russ

OlivierRRC committed:
Added staggered missile launch

LorenzoOrtizRRC committed:
- Fixed missing rigidbody variable when merging develop into this branch.

LorenzoOrtizRRC committed:
- projectile spawn point's up is now taken into account rather than rigidbody's transfom up.

dyfries committed:
Added ability manager

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Lorenzo' into develop

dyfries committed:
Merge remote-tracking branch 'remote/Dev_Lorenzo' into develop

Tsurei committed:
Update Blink

OlivierRRC committed:
Animated Spawning, Created Prefab

Quinton78 committed:
Adding meta files

OlivierRRC committed:
Attached windup animation to parent windup timer

OlivierRRC committed:
Merge branch 'develop' into Dev_Olivier

SteveLaird committed:
Ring Ability Added to Dev_Steven

Quinton78 committed:
Merge remote-tracking branch 'origin/develop' into Dev_Quentin

Quinton78 committed:
Added ability assets and script to my dev folder

Quinton78 committed:
Merge pull request #22 from dyfries/Dev_Quentin

Fixed merge conflicts from yesterday and moved assets and script into my dev folder

dyfries committed:
Merge pull request #21 from dyfries/Dev_Olivier

Missile Barrage Script

dyfries committed:
Lazer and Boost on the same player. Both are Ability subclasses that add functionality. Right now both will fire but we can add a system for "collecting" abilities soon.

dyfries committed:
Merge remote-tracking branch 'remote/mergingBranch' into Dev_Dylan

Updating physics layers and colors of boundaries.

# Conflicts:
#	Assets/2DSchmupp/Prefabs/Map/Boundaries.prefab

dyfries committed:
Merge branch 'develop' into Dev_Dylan

dyfries committed:
Ability

CraigFranceschetti committed:
Merge branch 'develop' of https://github.com/dyfries/SharedGameP1_23 into Dev_Craig

agamis28 committed:
Merge branch 'develop' into Dev_Angel

Tsurei committed:
Merge branch 'develop' into Dev_Russ

agamis28 committed:
Fixing spelling for invincibility ability script and scene.

Quinton78 committed:
Merge pull request #23 from dyfries/Dev_Quentin

dev_Quentin

CraigFranceschetti committed:
Changed physics layer to be playerAbility. Changed bomb starting location to be underneath the player when fired.

Tsurei committed:
Added task stubs to ability_blink

Tsurei committed:
Created testing area scene for ability_blink

agamis28 committed:
Making the invincibility ability exclude only the NPC layers from collision

agamis28 committed:
Adding more enemies in test scene and trying to make particle materials

alexbacon01 committed:
did some refactoring and added spots to put in animations

alexbacon01 committed:
added colour changes to show animation

CraigFranceschetti committed:
bomb now shoots forward for the duration of windup, and then blows up on action.

CraigFranceschetti committed:
Added comments to show how things are currently set up.

CraigFranceschetti committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into Dev_Craig

CraigFranceschetti committed:
Merge pull request #24 from dyfries/Dev_Craig

Added more functionality to the bomb

Tsurei committed:
Removed Empty test scene

Tsurei committed:
Created Test scene

Tsurei committed:
Added some variables that will be used in the future

LorenzoOrtizRRC committed:
- Added Cosmic Bombardment ability script.
- Added prefab for Cosmic Bombardment ability.
- Renamed Dev_Lorenzo test scene.

alexbacon01 committed:
added into the StopSlowDown function so that it adds force to the object again so it returns to previous speed

LorenzoOrtizRRC committed:
- Created blank GalacticCrusadeSpawner script.

LorenzoOrtizRRC committed:
- Renamed CosmicBombardment to GalacticCrusade.

Quinton78 committed:
Added comments in code to make it clear for future me what I need to do for the shield

Tsurei committed:
added animation controller to ability blink

LorenzoOrtizRRC committed:
- Added incomplete base functionality of Galactic Crusade Spawner.

Tsurei committed:
Modified animation controller for blink

Tsurei committed:
Created blink animation holders

Tsurei committed:
Update Blink_Test - Copy.unity

alexbacon01 committed:
converted the ability into a prefab, made adjustments to how the script deals with colliders to compensate for this.

LorenzoOrtizRRC committed:
- Finished base implementation of GalacticCrusadeSpawner.

CraigFranceschetti committed:
Added Coroutine to manually time when I should turn the collider off for the bomb.

CraigFranceschetti committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into Dev_Craig

CraigFranceschetti committed:
Merge pull request #25 from dyfries/Dev_Craig

Added Coroutine to manually time when I should turn the collider off …

SteveLaird committed:
Steven Demo Scene created, Second ability script 'HellFire' added with triangle projectile prefab.

LorenzoOrtizRRC committed:
- Added proper spawn area calculation into GalacticCrusade class.
- Integrated GalacticCrusadeSpawners to GalacticCrusade class.

LorenzoOrtizRRC committed:
- Finished base functionality of GalacticCrusade class.

LorenzoOrtizRRC committed:
- Added TODO comment.

LorenzoOrtizRRC committed:
- Created empty GalacticCrusadeShip class.

LorenzoOrtizRRC committed:
- GalacticCrusade ability's spawners array now reference spawners already made rather than un-instantiated prefabs.

LorenzoOrtizRRC committed:
- Removed coroutine related code in Galactic Crusade ability.

LorenzoOrtizRRC committed:
- Renamed Ability_CosmicBombardment prefab to Ability_GalacticCrusade.

Quinton78 committed:
Added key frames for shield animation just for testing

LorenzoOrtizRRC committed:
- Preparing to test GalacticCrusade ability by adding placeholder spawner and ship prefab.

Quinton78 committed:
Merge pull request #26 from dyfries/Dev_Quentin

Added key frames from shield animation for testing

Tsurei committed:
Merge branch 'Dev_Russ' into mergingBranch

Tsurei committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into mergingBranch

alexbacon01 committed:
puts a sprite over frozen objects to look like they're frozen in an ice block

alexbacon01 committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into Dev_Alex

# Conflicts:
#	ProjectSettings/TagManager.asset

alexbacon01 committed:
Merge pull request #28 from dyfries/Dev_Alex

Dev alex

LorenzoOrtizRRC committed:
- Tested GalacticCrusade ability.
- Added placeholder functionality to GalacticCrusadeShip.

LorenzoOrtizRRC committed:
- Fixed spawner random spawn point not being random. Syntax error.

agamis28 committed:
Taking out particle system because switching idea of ability a bit

agamis28 committed:
Merge branch 'mergingBranch' into Dev_Angel

SteveLaird committed:
HellFire ability outward movement fixed.

LorenzoOrtizRRC committed:
- small variable changes in test scene.

CraigFranceschetti committed:
Changing the ability to use a circle cast rather than a colldier.

Tsurei committed:
Linked Animations to blink prefab ability

SteveLaird committed:
Added list to track projectiles and state refactoring states to reference the list.

SteveLaird committed:
Merge branch 'Dev_Steven' into mergingBranch

Tsurei committed:
Added blink bubble collider

LorenzoOrtizRRC committed:
Merge branch 'Dev_Lorenzo' into mergingBranch

CraigFranceschetti committed:
Bomb now works with circle cast correctly.

LorenzoOrtizRRC committed:
Merge commit '2db25671bd8234155117e652cdc018b001cb8636' into Dev_Lorenzo

LorenzoOrtizRRC committed:
- Refactored bounds for more clarity.

LorenzoOrtizRRC committed:
Merge branch 'mergingBranch' into Dev_Lorenzo

Tsurei committed:
Added blink animation to prefab and scene

LorenzoOrtizRRC committed:
- Fixed bounds size syntax.

Tsurei committed:
adjusted prefab settings

Tsurei committed:
Merge branch 'Dev_Russ' into mergingBranch

CraigFranceschetti committed:
Merge pull request #29 from dyfries/UpdatingBomb

Changed the bomb to use a circle cast to meet requirements

Tsurei committed:
saved scene

Tsurei committed:
Merge branch 'Dev_Russ' into mergingBranch

Tsurei committed:
Fixed Player Blink animation

Tsurei committed:
Merge branch 'Dev_Russ' into mergingBranch

Tsurei committed:
edited blink animation speeds

Tsurei committed:
Merge branch 'Dev_Russ' into mergingBranch

agamis28 committed:
Adding Minion class. Will be instantiated by ability

SteveLaird committed:
Ability crosshair added for potential future functionalities.

agamis28 committed:
Adding minion into scene

SteveLaird committed:
Crosshair fixed, now a prefab. Projectile rotations implemented.

CraigFranceschetti committed:
Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into Dev_Craig

agamis28 committed:
Adding self destruct feature and changed rigidbody variable name that was causing issues

agamis28 committed:
Merge branch 'mergingBranch' into Dev_Angel

agamis28 committed:
Created a hit NPC function in minion script, that destroys an NPC when close

dyfries committed:
Added Ability Prefab as its own thing, with updated player (with ability prefab). Also added boundary collision layer.

dyfries committed:
Cleanup and organization of some art files.

dyfries committed:
Tried to reset to commit but this all got added here. What happened?

dyfries committed:
Note: I wiped out your collision matrix in the merge, sorry.

Merge branch 'mergingBranch' of https://github.com/dyfries/SharedGameP1_23 into mergingBranch

# Conflicts:
#	Assets/Devs/Dev_Angel/InvincibilityTestScene.unity
#	ProjectSettings/Physics2DSettings.asset
#	ProjectSettings/TagManager.asset

agamis28 committed:
Implemented instantiation of minions on ability

Created minion prefabs

agamis28 committed:
Cleaning up minion.cs code, and adding comments

dyfries committed:
Manually fixing up the last merge, which seemed to just bake the conflict directly into the asset. If you every enounter this, you can just edit the text files directly to remove the merge notes, as I have done here.

SteveLaird committed:
Sound effects added and implemented. Files organized. Crosshair in windup. State refactoring.

CraigFranceschetti committed:
Worked on the bomb particle system. Looks pretty nice, Needs SFX and the bomb needs a little animation before blowing up.

CraigFranceschetti committed:
Merge pull request #30 from dyfries/Dev_Craig

Bomb particle VFX and test scene.

agamis28 committed:
Gave minion a self destruct animation

SteveLaird committed:
New art and animations. Raycasts adjusted, Projectile collider and effects improved.

agamis28 committed:
Implemented background music, minion sound, and minion spawn sound.

Fixed null npc error. Imported audio sounds for scene. Fixed minon script null reference

SteveLaird committed:
Prefabs renamed. Method and Variable refactoring and organization, new Toggles. RayCast and OverlapCircle lock ons implemented.

SteveLaird committed:
Detection Range DrawRay Gizmo Fixed.

agamis28 committed:
Scaling explosion animation up and ajusting self destruct time

agamis28 committed:
Merge branch 'develop' into Dev_Angel

# Conflicts:
#	Assets/Devs/Dev_Angel/Minion.cs

agamis28 committed:
Merge remote-tracking branch 'origin/mergingBranch' into Dev_Angel

agamis28 committed:
Adding a collider to minion prefab

agamis28 committed:
Adding more sound effects, and making minion not move when exploding

agamis28 committed:
Stopping minions colliding together by excluding itself "player"

agamis28 committed:
Seperating audio sources

agamis28 committed:
Changing main camera size in Invincibility test scene

SteveLaird committed:
Null check for Lock On collision Physics2d layers fixed.

agamis28 committed:
Fixing spaceship moving sounds to only when the spaceship is moving

agamis28 committed:
Ajusting volumes and adding the invincibility start sound

agamis28 committed:
Organizing assets, and ajusting how sounds play on invincibility ability.

agamis28 committed:
Adding more NPCs and moving the boxes in scene

agamis28 committed:
Making varients of speed for minions that are spawned. Organizing hierarchy in invincibiliy scene

agamis28 committed:
Adding color variations to invincibility prefabs

agamis28 committed:
Adding a check for null references

SteveLaird committed:
Code refactoring and more null checks added. lockOnRotationSpeed implemented.

SteveLaird committed:
Comments added to Ability_HellFire script. Lowered prefab cooldown time.

SteveLaird committed:
Merge branch 'Dev_Steven' into mergingBranch

SteveLaird committed:
Music and engine sound added, sound effects changed, ship thruster animations added, explosion and crosshair parent fix.

SteveLaird committed:
Color state removed, Null check for ProjectilePrefab and ProjectileHolder added. NPC explosion enabled.

SteveLaird committed:
Merge branch 'Dev_Steven' into mergingBranch

agamis28 committed:
Adding asteroid sprites to obsticals and changed player that was active to the faster minions variation

agamis28 committed:
Merge branch 'mergingBranch' into Dev_Angel

agamis28 committed:
Added health system for invincibility player, as well as a collision detection system. Changed speed on faster variation of minons

agamis28 committed:
Set up animations for player dealth

agamis28 committed:
Set up background color and added border walls. Changed layers to allow player to properly become invinsible. Also changed NPC prefab to exclude the walls from collision.

agamis28 committed:
Added a health bar controller. Added player prefab. Activated main variation of player.

Only needs audio for getting damaged, and dying now.

agamis28 committed:
Adding collision sounds

agamis28 committed:
Making sure the player doesn't take damage or collide with enviorment when invincible

dyfries committed:
Basic examples of 2D style particle systems.

dyfries committed:
Merge branch 'mergingBranch' into develop

# Conflicts:
#	ProjectSettings/TagManager.asset

SteveLaird committed:
Target/Crosshair position now a vector, includes wire sphere gizmo for it's position in editor.

SteveLaird committed:
Ability speed and rotation variable ranges increased. Scene Fix. Layers Fix.

SteveLaird committed:
Layers and Lists fixed.

SteveLaird committed:
Layers fixed. Missed in last commit.

SteveLaird committed:
Firing Pause, Projectile Grow & Shrink, Moving Background, Reload Animation

SteveLaird committed:
Update method basic refactor, some added comments. Changed state check for reload animation.

SteveLaird committed:
Set All Variables to Private. Presets Made.

SteveLaird committed:
Scene Cleanup and Folder Organization. NPC reference fixed for its effects.
